lib/file.js

```diff
@@ -42,3 +43,26 @@ exports.writeToFile = function (path, data) {
     }
   });
 };
+
+/**
+ * Normalize the input/output file path according to a sub-directory if needed
+ * @param {String} file - path of the input/ouput file
+ * @param {String|null} subDirectory - an explicit path to a sub-directory if given
+ * @returns {Promise}
+ */
+exports.normalizeFilePath = function (file, subDirectory) {
+  return Bluebird.resolve()
```

Comment:
It doesn't look like anything in this function is actually asynchronous so you don't have to make it return a promise. It should make the implementation a bit simpler.